app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt

```diff
@@ -78,6 +78,15 @@ class LoginFragment : Fragment() {
             loginActivityViewModel.sendResetPasswordEmail(email.text.toString())
         }
 
+        loginActivityViewModel.user.observe(this, Observer {
+            redirectToMain()
+        })
```

Comment:
Is this needed?

Reply:
Yeah, actually, this will ensure that the user is redirected to the main page only after the user is fetched succesfully. Just as the user is fetched, it will then redirect to the main activity or else if I would have just used the `redirectToMain()`function in the loggedIn observer, the user would be redirected to the main page just after logging in, thus fetchProfile() won't be of any use :+1: